core/transcoder_test.go

```diff
@@ -59,19 +59,86 @@ func TestNvidiaTranscoder(t *testing.T) {
 		t.Skip("No device specified; skipping remainder of Nvidia tests")
 		return
 	}
-	tc = NewNvidiaTranscoder(dev, tmp)
+	StartNvidiaTranscoders(dev, tmp)
+	tc = NewNvidiaTranscoder(dev)
 	res, err := tc.Transcode("", fname, profiles)
 	if err != nil {
 		t.Error(err)
 	}
-	if Over1Pct(len(res.Segments[0].Data), 650292) {
+	if Over1Pct(len(res.Segments[0].Data), 485416) {
 		t.Errorf("Wrong data %v", len(res.Segments[0].Data))
 	}
-	if Over1Pct(len(res.Segments[1].Data), 884164) {
+	if Over1Pct(len(res.Segments[1].Data), 771740) {
 		t.Errorf("Wrong data %v", len(res.Segments[1].Data))
 	}
 }
 
+func TestNvidia_Stack(t *testing.T) {
+	assert := assert.New(t)
+	ss := newSegStack()
+	assert.Empty(ss.segs, "Sanity check for empty stack")
+	for i := 0; i < 1000; i++ {
+		fname := fmt.Sprintf("%d", i)
+		ss.push(&nvSegData{fname: fname})
+	}
+	for i := 999; i >= 0; i-- {
+		fname := fmt.Sprintf("%d", i)
+		seg := ss.pop()
+		assert.Equal(fname, seg.fname)
+	}
+	assert.Empty(ss.segs, "Stack nonempty") // sanity check
+
+	// "Check" popping an empty stack blocks. Approximate this with a timeout
+	wg := newWg(1)
+	go func() {
+		ss.pop()
+		wg.Done()
+	}()
+	assert.False(wgWait2(wg, 100*time.Millisecond), "Did not time out on empty stack")
+}
+
+func TestNvidia_ConcurrentStack(t *testing.T) {
+	// Run this under `-race`
+	//assert := assert.New(t)
```

Comment:
Can remove this.

Reply:
Fixed in 26c1f19b46bb9546bdbce22cbe3f890d957e0310